### PR TITLE
full review pre-trial 03

### PR DESCRIPTION
**Pull request by avallecam**
This PR aims to collect edit suggestions before trial 03.

Rendered website: https://epiverse-trace.github.io/tutorials-late/

We encourage in-line specific changes in the files changed tab (read [GitHub how-to guide steps 5 to 7](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request))

### CODE REVIEW

File: episodes/modelling-interventions.Rmd [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+  theme_bw(
+    base_size = 15
+  ) +
+  geom_vline(
+    xintercept = c(mask_mandate$time_begin, mask_mandate$time_end),
+    colour = "black",
+    linetype = "dashed",
+    linewidth = 0.2
+  ) +
+  annotate(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:280]:
```suggestion
  ggplot2::annotate(
```

File: episodes/modelling-interventions.Rmd [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+  time_end = 300, increment = 1.0
+)
+
+# create intervention_type column for plotting
+output_school$intervention_type <- "school closure"
+output_baseline$intervention_type <- "baseline"
+output <- rbind(output_school, output_baseline)
+
+ggplot(data = output[output$compartment == "infectious", ]) +
+  aes(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:172]:
```suggestion
  ggplot2::aes(
```

File: episodes/modelling-interventions.Rmd [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+    x = time,
+    y = value,
+    color = intervention_type,
+    linetype = intervention_type
+  ) +
+  stat_summary(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:178]:
```suggestion
  ggplot2::stat_summary(
```

File: episodes/modelling-interventions.Rmd [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+    fun = "sum",
+    geom = "line",
+    linewidth = 1
+  ) +
+  scale_y_continuous(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:183]:
```suggestion
  ggplot2::scale_y_continuous(
```

File: episodes/modelling-interventions.Rmd [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+    labels = scales::comma
+  ) +
+  labs(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:186]:
```suggestion
  ggplot2::labs(
```

File: episodes/modelling-interventions.Rmd [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+    x = "Simulation time (days)",
+    y = "Individuals"
+  ) +
+  theme_bw(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:190]:
```suggestion
  ggplot2::theme_bw(
```

File: episodes/modelling-interventions.Rmd [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+    base_size = 15
+  ) +
+  geom_vline(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:193]:
```suggestion
  ggplot2::geom_vline(
```

File: episodes/modelling-interventions.Rmd [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+    y = 10,
+    angle = 0,
+    vjust = "outward"
+  )
+```
+
+::::::::::::::::::::::::::::::::::::: callout
+### Intervention types
+
+There are two intervention types for `model_default()`. Rate interventions on model parameters (`transmission_rate` $\beta$, `infectiousness_rate` $\sigma$ and `recovery_rate` $\gamma$) and contact matrix reductions `contacts`.
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:295]:
```suggestion
There are two intervention types for `model_default()`. Rate interventions on model parameters (`transmission_rate` $\beta$, `infectiousness_rate` $\sigma$ and `recovery_rate` $\gamma$) and contact matrix reductions (`contacts`).
```

File: episodes/modelling-interventions.Rmd [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+
+To implement both contact and rate interventions in the same simulation they must be passed as a list e.g. `intervention = list(transmission_rate = mask_mandate, contacts = close_schools)`. But if there are multiple interventions that target contact rates, these must be passed as one `contacts` input. See the [vignette on modelling overlapping interventions](https://epiverse-trace.github.io/epidemics/articles/multiple_interventions.html) for more detail.
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:297]:
```suggestion
To implement both contact and rate interventions in the same simulation they must be passed as a list, e.g., `intervention = list(transmission_rate = mask_mandate, contacts = close_schools)`. But if there are multiple interventions that target contact rates, these must be passed as one `contacts` input. See the [vignette on modelling overlapping interventions](https://epiverse-trace.github.io/epidemics/articles/multiple_interventions.html) for more detail. 
```

File: episodes/modelling-interventions.Rmd [hunk 10]
```diff
@@ -0,0 +1,442 @@
...
+
+To implement both contact and rate interventions in the same simulation they must be passed as a list e.g. `intervention = list(transmission_rate = mask_mandate, contacts = close_schools)`. But if there are multiple interventions that target contact rates, these must be passed as one `contacts` input. See the [vignette on modelling overlapping interventions](https://epiverse-trace.github.io/epidemics/articles/multiple_interventions.html) for more detail. 
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+## Pharmaceutical interventions
+
+Pharmaceutical interventions (PIs) are measures such as vaccination and mass treatment programs. In the previous section, we assumed that interventions reduced the value of parameter values while the intervention was in place. In the case of vaccination, we assume that after the intervention individuals are no longer susceptible and should be classified into a different disease state. Therefore, we specify the rate at which individuals are vaccinated and track the number of vaccinated individuals through time.
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:304]:
```suggestion
Pharmaceutical interventions (PIs) are measures such as vaccination and mass treatment programs. In the previous section, we integrated the interventions into the model by reducing  parameter values during specific period of time window in which these intervention set to take place. In the case of vaccination, we assume that after the intervention individuals are no longer susceptible and should be classified into a different disease state. Therefore, we specify the rate at which individuals are vaccinated and track the number of vaccinated individuals over time. 
```

File: episodes/modelling-interventions.Rmd [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+$$
+\begin{aligned}
+\frac{dS_i}{dt} & = - \beta S_i \sum_j C_{i,j} I_j -\nu_{t} S_i \\
+\frac{dE_i}{dt} &= \beta S_i\sum_j C_{i,j} I_j - \alpha E_i \\
+\frac{dI_i}{dt} &= \alpha E_i - \gamma I_i \\
+\frac{dR_i}{dt} &=\gamma I_i \\
+\frac{dV_i}{dt} & =\nu_{i,t} S_i\\
+\end{aligned}
+$$
+Individuals are vaccinated at an age group ($i$) specific time dependent ($t$) vaccination rate ($\nu_{i,t}$). The SEIR components of these equations are described in the tutorial [simulating transmission](../episodes/simulating-transmission.md#simulating-disease-spread).
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:353]:
```suggestion
Individuals in age group ($i$)  at specific time dependent ($t$)  are vaccinated at rate ($\nu_{i,t}$). The other SEIR components of these equations are described in the tutorial [simulating transmission](../episodes/simulating-transmission.md#simulating-disease-spread). 
```

File: episodes/modelling-interventions.Rmd [hunk 12]
```diff
@@ -0,0 +1,442 @@
...
+\frac{dS_i}{dt} & = - \beta S_i \sum_j C_{i,j} I_j -\nu_{t} S_i \\
+\frac{dE_i}{dt} &= \beta S_i\sum_j C_{i,j} I_j - \alpha E_i \\
+\frac{dI_i}{dt} &= \alpha E_i - \gamma I_i \\
+\frac{dR_i}{dt} &=\gamma I_i \\
+\frac{dV_i}{dt} & =\nu_{i,t} S_i\\
+\end{aligned}
+$$
+Individuals are vaccinated at an age group ($i$) specific time dependent ($t$) vaccination rate ($\nu_{i,t}$). The SEIR components of these equations are described in the tutorial [simulating transmission](../episodes/simulating-transmission.md#simulating-disease-spread). 
+
+To explore the effect of vaccination we need to create a vaccination object to pass as an input into `model_default()` that includes an age groups specific vaccination rate `nu` and age group specific start and end times of the vaccination program (`time_begin` and `time_end`).
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:355]:
```suggestion
To explore the effect of vaccination we need to create a vaccination object to pass as an input into `model_default()` that includes  age groups specific vaccination rate `nu` and age groups specific start and end times of the vaccination program (`time_begin` and `time_end`). 
```

File: episodes/modelling-interventions.Rmd [hunk 13]
```diff
@@ -0,0 +1,442 @@
...
+\frac{dI_i}{dt} &= \alpha E_i - \gamma I_i \\
+\frac{dR_i}{dt} &=\gamma I_i \\
+\frac{dV_i}{dt} & =\nu_{i,t} S_i\\
+\end{aligned}
+$$
+Individuals are vaccinated at an age group ($i$) specific time dependent ($t$) vaccination rate ($\nu_{i,t}$). The SEIR components of these equations are described in the tutorial [simulating transmission](../episodes/simulating-transmission.md#simulating-disease-spread). 
+
+To explore the effect of vaccination we need to create a vaccination object to pass as an input into `model_default()` that includes an age groups specific vaccination rate `nu` and age group specific start and end times of the vaccination program (`time_begin` and `time_end`). 
+
+Here we will assume all age groups are vaccinated at the same rate 0.01 and that the vaccination program starts on day 40 and is in place for 150 days.
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:357]:
```suggestion
Here we will assume all age groups are vaccinated at the same rate 0.01 and that the vaccination program starts on day 40 and continue to be in place for 150 days.
```

File: episodes/modelling-interventions.Rmd [hunk 14]
```diff
@@ -0,0 +1,442 @@
...
+
+```{r vaccinate}
+# prepare a vaccination object
+vaccinate <- vaccination(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:361]:
```suggestion
vaccinate <- epidemics::vaccination(
```

File: episodes/modelling-interventions.Rmd [hunk 15]
```diff
@@ -0,0 +1,442 @@
...
+  time_begin = matrix(40, nrow(contact_matrix)),
+  time_end = matrix(40 + 150, nrow(contact_matrix)),
+  nu = matrix(c(0.01, 0.01, 0.01))
+)
+```
+
+We pass our vaccination object using `vaccination = vaccinate`:
+
+```{r output_vaccinate}
+output_vaccinate <- model_default(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:372]:
```suggestion
output_vaccinate <- epidemics::model_default(
```

File: episodes/modelling-interventions.Rmd [hunk 16]
```diff
@@ -0,0 +1,442 @@
...
+# prepare a vaccination object
+vaccinate <- vaccination(
+  name = "vaccinate all",
+  time_begin = matrix(40, nrow(contact_matrix)),
+  time_end = matrix(40 + 150, nrow(contact_matrix)),
+  nu = matrix(c(0.01, 0.01, 0.01))
+)
+```
+
+We pass our vaccination object using `vaccination = vaccinate`:
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:369]:
```suggestion
We pass our vaccination object into the model using argument `vaccination = vaccinate`:
```

File: episodes/modelling-interventions.Rmd [hunk 17]
```diff
@@ -0,0 +1,442 @@
...
+:::::::::::::::::::::::: solution 
+
+## Output
+
+```{r plot_vaccinate, echo = TRUE, message = FALSE, fig.width = 10}
+# create intervention_type column for plotting
+output_vaccinate$intervention_type <- "vaccination"
+output <- rbind(output_school, output_masks, output_vaccinate, output_baseline)
+
+ggplot(data = output[output$compartment == "infectious", ]) +
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:399]:
```suggestion
ggplot2::ggplot(data = output[output$compartment == "infectious", ]) +
```

File: episodes/modelling-interventions.Rmd [hunk 18]
```diff
@@ -0,0 +1,442 @@
...
+  aes(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:400]:
```suggestion
  ggplot2::aes(
```

File: episodes/modelling-interventions.Rmd [hunk 19]
```diff
@@ -0,0 +1,442 @@
...
+    x = time,
+    y = value,
+    color = intervention_type,
+    linetype = intervention_type
+  ) +
+  stat_summary(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:406]:
```suggestion
  ggplot2::stat_summary(
```

File: episodes/modelling-interventions.Rmd [hunk 20]
```diff
@@ -0,0 +1,442 @@
...
+    fun = "sum",
+    geom = "line",
+    linewidth = 1
+  ) +
+  scale_y_continuous(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:411]:
```suggestion
  ggplot2::scale_y_continuous(
```

File: episodes/modelling-interventions.Rmd [hunk 21]
```diff
@@ -0,0 +1,442 @@
...
+    labels = scales::comma
+  ) +
+  labs(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:414]:
```suggestion
  ggplot2::labs(
```

File: episodes/modelling-interventions.Rmd [hunk 22]
```diff
@@ -0,0 +1,442 @@
...
+    x = "Simulation time (days)",
+    y = "Individuals"
+  ) +
+  theme_bw(
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:418]:
```suggestion
  ggplot2::theme_bw(
```

File: episodes/modelling-interventions.Rmd [hunk 23]
```diff
@@ -0,0 +1,442 @@
...
+
+:::::::::::::::::::::::::::::::::
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+
+
+## Summary
+
+Different types of intervention can be implemented using mathematical modelling. Modelling interventions requires assumptions of which model parameters are affected (e.g. contact matrices, transmission rate), by what magnitude and and what times in the simulation of an outbreak.
```
Comment (Degoot-AM) [inline episodes/modelling-interventions.Rmd:433]:
```suggestion
Different types of intervention can be implemented using mathematical modelling. Modelling interventions requires assumptions of which model parameters are affected (e.g. contact matrices, transmission rate), and by what magnitude  and what times in the simulation of an outbreak. 
```

**Review by Degoot-AM — CHANGES_REQUESTED** [review]

**Comment by Degoot-AM** [comment]
@avallecam, I enjoyed reading this tutorials and suggest a number of changes to be made. Below are my general remarks:

-   My review did not include the `comparing scenario` as it yet to be completed. 
-  I found this tutorial a bit lengthy, you may consider breaking it down
-  Fail to generate the in-fly transmission diagrams embedded with episodes locally in machine using the workbench framework. I got the following error message `Error in (function (url = NULL, file = "webshot.png", vwidth = 992, vheight = 744, : webshot.js returned failure value: 1`. I do not  know whether this a general problem or specific to my machine, but is worth to check it out. 
-  Use `tracetheme` package in the making the graphs.
- Link the tutorials together to provide a smooth order to flaw.

File: episodes/simulating-transmission.Rmd [hunk 24]
```diff
@@ -0,0 +1,555 @@
...
+# build for all age groups
+initial_conditions <- rbind(
+  initial_conditions_inf,
+  initial_conditions_free,
+  initial_conditions_free
+)
+rownames(initial_conditions) <- rownames(contact_matrix)
+
+# prepare the population to model as affected by the epidemic
+uk_population <- population(
```
[the remark this answers is not in the document]
Comment (avallecam) [reply episodes/simulating-transmission.Rmd:88]: appropriate edit suggestion!

File: episodes/simulating-transmission.Rmd [hunk 25]
```diff
@@ -0,0 +1,555 @@
...
+To create our population object, we call the function `population()` specifying a name, the contact matrix, the demography vector and the initial conditions.
+
+```{r population}
+uk_population <- population(
+  name = "UK",
+  contact_matrix = contact_matrix,
+  demography_vector = demography_vector,
+  initial_conditions = initial_conditions
+)
+```
```
Comment (avallecam) [inline episodes/simulating-transmission.Rmd:353]:
```suggestion
To create our population object, from the {epidemics} package we call the function `population()` specifying a name, the contact matrix, the demography vector and the initial conditions.

```{r population}
uk_population <- epidemics::population(
  name = "UK",
  contact_matrix = contact_matrix,
  demography_vector = demography_vector,
  initial_conditions = initial_conditions
)
```
```

File: episodes/simulating-transmission.Rmd [hunk 26]
```diff
@@ -0,0 +1,555 @@
...
+- infectiousness rate $\alpha$ (preinfectious period=$1/\alpha$),
+- recovery rate $\gamma$ (infectious period=$1/\gamma$).
+
+In `epidemics`, we specify the model inputs as :
+
+- `transmission_rate` = $R_0 \gamma$,
+- `infectiousness_rate` = $\alpha$,
+- `recovery_rate` = $\gamma$, 
+  
+We will simulate a strain of influenza with pandemic potential with $R_0=1.46$, a preinfectious period of 3 days and infectious period of 7 days. Therefore our inputs will be:
```
Comment (avallecam) [inline episodes/simulating-transmission.Rmd:370]: Which one is clearer? `latent` period or `preinfectious` period?

```suggestion
We will simulate a strain of influenza with pandemic potential with $R_0=1.46$, a preinfectious period of 3 days and infectious period of 7 days.

- `latent_period <- 3.0`
- `infectious_period <- 7.0`
- `basic_reproduction <- 1.46`

Therefore our inputs will be:
```

File: episodes/simulating-transmission.Rmd [hunk 27]
```diff
@@ -0,0 +1,555 @@
...
+
+- `transmission_rate = 1.46 / 7.0`,
+- `infectiousness_rate = 1.0 / 3.0`,
+- `recovery_rate = 1.0 / 7.0`.
```
Comment (avallecam) [inline episodes/simulating-transmission.Rmd:374]: For consisteny with the callout box bellow

```suggestion
- `infectiousness_rate = 1/latent_period`,
- `recovery_rate = 1/infectious_period`,
- `transmission_rate =  basic_reproduction*recovery_rate`.
```

File: episodes/simulating-transmission.Rmd [hunk 28]
```diff
@@ -0,0 +1,555 @@
...
+)
+head(output)
+```
+
+*Note : This model also has the functionality to include vaccination and tracks the number of vaccinated individuals through time. Even though we have not specified any vaccination, there is still a vaccinated compartment in the output (containing no individuals). We will cover the use of vaccination in future tutorials.*
+
+Our model output consists of the number of individuals in each compartment in each age group through time. We can visualise the infectious individuals only (those in the $I$ class) through time.
+
+```{r visualise, fig.width = 10}
+filter(output_plot, compartment %in% c("exposed", "infectious")) %>%
```
Comment (avallecam) [inline episodes/simulating-transmission.Rmd:426]:
```suggestion
library(tidyverse)

filter(output, compartment %in% c("exposed", "infectious")) %>%
```

**Review by avallecam — COMMENTED** [review]
the term preinfectious is used across the episodes. Is there a preference to use `preinfectious` over `latent` period?

**Review by avallecam — COMMENTED** [review]
brief link edits

File: episodes/modelling-interventions.Rmd [hunk 29]
```diff
@@ -0,0 +1,442 @@
...
+    vjust = "outward"
+  )
+```
+
+::::::::::::::::::::::::::::::::::::: callout
+### Intervention types
+
+There are two intervention types for `model_default()`. Rate interventions on model parameters (`transmission_rate` $\beta$, `infectiousness_rate` $\sigma$ and `recovery_rate` $\gamma$) and contact matrix reductions (`contacts`).
+
+To implement both contact and rate interventions in the same simulation they must be passed as a list, e.g., `intervention = list(transmission_rate = mask_mandate, contacts = close_schools)`. But if there are multiple interventions that target contact rates, these must be passed as one `contacts` input. See the [vignette on modelling overlapping interventions](https://epiverse-trace.github.io/epidemics/articles/multiple_interventions.html) for more detail.
```
Comment (avallecam) [inline episodes/modelling-interventions.Rmd:297]:
```suggestion
To implement both contact and rate interventions in the same simulation they must be passed as a list, e.g., `intervention = list(transmission_rate = mask_mandate, contacts = close_schools)`. But if there are multiple interventions that target contact rates, these must be passed as one `contacts` input. See the [vignette on modelling overlapping interventions](https://epiverse-trace.github.io/epidemics/articles/modelling_multiple_interventions.html) for more detail. 
```

**Comment by avallecam** [comment]
closing as redirected to main in #29

**Pull request closed without merging**